File: Java/Problem1.java
import java.io.*;
import java.util.*;

class Problem1 {
    public static void main(String args[]){
        File file = new File("/home/ec2-user/environment/AOC/Resources/problem1.txt");
        
        try{
            System.out.println("Part 1: "+part1(file));
            
            System.out.println("Part 2: "+part2(file));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    
    public static int part1(File file) throws Exception{
        BufferedReader br = new BufferedReader(new FileReader(file));
        String current; 
        HashSet<Integer> s = new HashSet<Integer>();
        while ((current = (br.readLine())) != null) {
            Integer i = Integer.parseInt(current);
            if(s.contains(2020-i)){
                return i * (2020-i);
            } else {
                s.add(i);
            }
        } 
        
        return -1;
    }
    
    public static int part2(File file) throws Exception{
        BufferedReader br = new BufferedReader(new FileReader(file));
        String current; 
        HashSet<Integer> s = new HashSet<Integer>();
        List<Integer> l = new ArrayList<Integer>();
        while ((current = (br.readLine())) != null) {
            Integer a = Integer.parseInt(current);
            s.add(a);
            l.add(a);
        } 
        
        Integer x,y;
        for(int i=0; i<l.size(); i++){
            for(int j=i+1; j<l.size(); j++){
                x = l.get(i);
                y = l.get(j);
                if(s.contains(2020 - x - y)){
                    return x * y * (2020 - x - y);
                }
            }
        }
        return -1;
    }
}

File: 2020/Python/problem1.py

def part1():
    f = open("/home/ec2-user/environment/AOC/Resources/problem1.txt")
    s = set()
    current = f.readline().strip()
    currentInt = 0
    answer = 0
    while(current is not None):
        currentInt = int(current)
        if((2020 - currentInt) not in s):
            s.add(currentInt)
        else:
            return currentInt * (2020-currentInt)
        current = f.readline().strip()
        
print(part1())

def part2():
    f = open("/home/ec2-user/environment/AOC/Resources/problem1.txt")
    s = set()
    values = []
    current = f.readline().strip()
    while(current is not None and current != ""):
        s.add(int(current))
        values.append(int(current))
        current = f.readline().strip()
    
    for i in range(len(values)):
        for j in range(len(values)):
            if(i != j and (2020 - values[i] - values[j]) in s):
                return values[i] * values[j] * (2020 - values[i] - values[j])

print(part2())
                
        